openedx/edx-proctoring
# fix: do not use course's due date when a timed exam has one specified

**Description:**

24199b466 introduced course end date check to allow viewing timed exams after course end date in self-paced courses. However, this change also means that the students cannot submit exams once the course due date is in the past. The exam's due date should have a higher priority so that the students with an individual due date extension (set through the Instructor Dashboard) can still complete an exam in such a case.

**Testing instructions**

1. [Set](http://localhost:18010/settings/details/course-v1:edX+DemoX+Demo_Course) the course's end date to the past.
1. [Set up](https://edx.readthedocs.io/projects/open-edx-building-and-running-a-course/en/latest/course_features/timed_exams.html#enable-timed-exams) a timed exam in a course. Set its due date to the future.
1. Sign in as an audit user and try to access the exam. If you do this point without the change from this PR, you will get a message that the exam has expired. With this change, the exam should be working correctly.

**Pre-Merge Checklist:**

- [x] Updated the version number in `edx_proctoring/__init__.py` and `package.json` if these changes are to be released.
- [x] Described your changes in `CHANGELOG.rst`
- [ ] Confirmed Github reports all automated tests/checks are passing.
- [ ] Approved by at least one additional reviewer.

**Post-Merge:**

- [ ] Create a tag matching the new version number.

**Other information**

Private Jira link: [BB-6658](https://tasks.opencraft.com/browse/BB-6658)
